minions/async/src/main/java/io/github/microcks/minion/async/AsyncAPITestManager.java

```diff
@@ -281,15 +281,26 @@ private String getExpectedContentType(JsonNode specificationNode, String message
          // Get message real content type if defined.
          String contentType = defaultContentType;
          JsonNode messageNode = specificationNode.at(messagePathPointer);
+
+         // messageNode will be an array of messages
+         if (messageNode.isArray() && messageNode.size() > 0) {
+            messageNode = messageNode.get(0);
+         }
+
          // If it's a $ref, then navigate to it.
-         if (messageNode.has("$ref")) {
+         while (messageNode.has("$ref")) {
             // $ref: '#/components/messages/lightMeasured'
             String ref = messageNode.path("$ref").asText();
-            messageNode = messageNode.at(ref.substring(1));
+            messageNode = specificationNode.at(ref.substring(1));
          }
          if (messageNode.has("contentType")) {
             contentType = messageNode.path("contentType").asText();
          }
+         if (messageNode.has("payload")) {
```

Comment:
IIRC `payload` is mandatory in both AsyncAPI 2.x and 3.x. I think this is not necessary. Remember that spec has been correctly parsed before during the import, so basic stuff like this shouldn't happen.